Sources/_Datadog_Private/ObjcAppLaunchHandler.m

```diff
@@ -56,32 +59,60 @@ @interface AppLaunchHandler : NSObject
 @implementation AppLaunchHandler
 
 + (void)load {
+    pthread_rwlock_init(&rwLock, NULL);
+
     // This is called at the `_Datadog_Private` load time, keep the work minimal
     FrameworkLoadTime = CFAbsoluteTimeGetCurrent();
 
     NSNotificationCenter * __weak center = NSNotificationCenter.defaultCenter;
-    id __block token = [center
-                        addObserverForName:UIApplicationDidBecomeActiveNotification
-                        object:nil
-                        queue:NSOperationQueue.mainQueue
-                        usingBlock:^(NSNotification *_){
-
-        pthread_rwlock_init(&rwLock, NULL);
+    id __weak __block token = [center addObserverForName:UIApplicationDidBecomeActiveNotification
+                                                  object:nil
+                                                   queue:NSOperationQueue.mainQueue
+                                              usingBlock:^(NSNotification *_){
         pthread_rwlock_wrlock(&rwLock);
-        TimeToApplicationDidBecomeActive = ComputeProcessTimeFromStart();
+        UIApplicationDidBecomeActiveNotificationTime = CFAbsoluteTimeGetCurrent();
         pthread_rwlock_unlock(&rwLock);
-
         [center removeObserver:token];
-        token = nil;
     }];
 }
 
 @end
 
+@implementation UIApplication (Tracking)
+
++ (void)load {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        method_exchangeImplementations(
+            class_getInstanceMethod(self, @selector(init)),
+            class_getInstanceMethod(self, @selector(dd_init))
+        );
+    });
+}
+
+- (instancetype)dd_init {
+    pthread_rwlock_wrlock(&rwLock);
+    UIApplicationInstantiateSingletonTime = CFAbsoluteTimeGetCurrent();
+    pthread_rwlock_unlock(&rwLock);
```

Comment:
I don't think we actually need to acquire the lock here, this should get invoke long before we need to read `UIApplicationInstantiateSingletonTime`